lib/semantics/check-omp-structure.cc

```diff
@@ -161,9 +162,10 @@ void OmpStructureChecker::Leave(const parser::OmpClauseList &) {
           const auto &collapseClause{
               std::get<parser::OmpClause::Collapse>(clause2->u)};
           // ordered and collapse both have parameters
-          const auto orderedValue{GetPosIntValue(orderedClause.v)};
-          const auto collapseValue{GetPosIntValue(collapseClause.v)};
-          if (orderedValue && collapseValue &&
+          const auto orderedValue{GetIntValue(orderedClause.v)};
+          const auto collapseValue{GetIntValue(collapseClause.v)};
+          if (orderedValue && collapseValue && orderedValue.value() > 0 &&
+              collapseValue.value() > 0 &&
               orderedValue.value() < collapseValue.value()) {
```

Comment:
I don't think you need the two `> 0` checks.

Reply:
If one or both of them are `<= 0`, it might not be necessary to throw the third error msg:
```
do.f90:4:12: error: The parameter of the COLLAPSE clause must be a constant positive integer expression
    !$omp do collapse(-1) ordered(-2)
             ^^^^^^^^^^^^
do.f90:4:25: error: The parameter of the ORDERED clause must be a constant positive integer expression
    !$omp do collapse(-1) ordered(-2)
                          ^^^^^^^^^^^
do.f90:4:25: error: The parameter of the ORDERED clause must be greater than or equal to the parameter of the COLLAPSE clause
    !$omp do collapse(-1) ordered(-2)
                          ^^^^^^^^^^^
f18: semantic errors in do.f90
```

Reply:
I think that Tim meant that you don't need to check for `collapseValue.value() > 0` since the next relation would fail anyway if it were not true.